### DISCUSSION TITLE
Changing video quality from auto to a fixed quality does not do anything. It's still auto.

### DISCUSSION THREAD

**Discussion by Ainson** [f1multiviewer/issue-tracker#130] [Q&A] [answered]
The video quality jumps to whatever it wants (auto) even if I set it to a fixed value.

Version 1.5.4, Windows 7 (64 bit)

**Comment by f1multiviewer** [comment, marked as answer]
Can you try setting it using the Context menu? So right-click > Set Video Quality > (pick your quality preference)? The player settings window is definitely buggy, not much I can do about that unfortunately.

**Comment by Ainson** [comment]
Doesn't seem to make a difference

**Reply by f1multiviewer** [reply]
What do you mean with "Doesn't seem to make a difference"? If you open the debug data (press `D`), do you see the resolution change? It may take a bit of time, as it doesn't discard anything that was buffered before.

It works fine when testing on my Windows VM:
<img width="926" alt="image" src="https://user-images.githubusercontent.com/106821994/180766642-7725c504-a805-4399-9bb7-e1b91c895aea.png">

**Comment by Ainson** [comment]
I meant it made no difference to the issue to set the quality via the context menu, rather than via the player settings box as was the suggestion.

The video quality might go up to the desired set quality and seems to work for a bit, but then drops back again (usually after switching between full screen and back), and it's the same whether it was set via the player settings, or via the context menu.

**Reply by f1multiviewer** [reply]
So it works (in contrast to "does not do anything"), but then later, falls back to `auto`? If so, it might be a network bandwidth constraint, I'm not 100% sure. You could look at the debug data, and look at the buffer length, if that drops below what is configured in settings for some time, it may override your original bandwidth selection. I'm not 100% sure, as this is behaviour defined by Bitmovin, the player used, which is closed-source, so hard to debug.

**Reply by Ainson** [reply]
Works as in 'stuck in auto'. Manual override does nothing, no matter how it's set.

It seems the issue is related to switching from small window to full screen and back (which is what I do all the time (to see the data channel)). Maybe the player thinks I don't need more res when it's small, and switches to a lower resolution to save bandwidth (even if not necessary).

My bandwidth is not an issue, I forgot to say. I have 100 mbs tested, and this is never an issue with F1TV in the browser.

Yeah, so looks to be with the player.

**Reply by f1multiviewer** [reply]
Ohh, you have a player that's smaller than 720p for OBC/1080p for others? Then it's definitely the configuration. I currently have a configuration entry that limits the resolution to the player size. I'll see if this can be configurable or just disabled.

**Reply by Ainson** [reply]
That would be great

**Comment by Ainson** [comment]
This worked fine for this race, so seems to be fixed.

Thanks. I already got you some coffees.

**Reply by f1multiviewer** [reply]
Great! I did introduce a setting into settings:

<img width="987" alt="image" src="https://user-images.githubusercontent.com/106821994/182047479-7f8b4ba1-14f9-45b0-9a74-26a7344577d5.png">

You'll want to turn this *off*

**Reply by recursiveGecko** [reply]
@f1multiviewer I'm not sure if you've received any other feedback about this, but I only _just_ discovered that this setting exists. It might be worth considering whether this should be turned off by default? I found it quite frustrating that my resolution choice wasn't being respected until I came across this thread, for the longest time I assumed it was a bug with the video player.
